File: TypeScript/Arrays/Search/LinearSearch.ts
const LinearSearch = (arr: number[], searchNumber: number): void => {
  if (arr == null || arr.length <= 1) {
    return;
  }
  let numFound: boolean = false;
  let index: number = 0;
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] === searchNumber) {
      numFound = true;
      index = i;
    }
  }
  if (numFound) {
    console.log(`The number is found at the index of ${index + 1}`);
  } else {
    console.log("404- Not found.");
  }
};

let arr: number[] = [32, 43, 5, 3, 2, 34233, 23];
LinearSearch(arr, 23);

File: Java/Arrays/Search/LinearSearch.java
package Search;

public class LinearSearch {

    public static void linearSearch(int arr[], int SearchNumber) {
        boolean numFound = false;
        int index = 0;
        if (arr.length == 0 || arr == null) {
            return;
        }
        for (int i = 0; i < arr.length; i++) {
            if (SearchNumber == arr[i]) {
                numFound = true;
                index = i;
                break;
            }
        }
        if (numFound) {
            System.out.println("The number is found at the index of " + (index + 1));
        } else {
            System.out.println("Not found.");
        }
    }

    public static void main(String[] args) {
        int arrNum[] = { 32, 4, 2, 55, 634, 23, 2 };
        int searchNumber = 23;
        linearSearch(arrNum, searchNumber);
    }
}